inst/rmarkdown/templates/rjournal_article/resources/template.tex

```diff
@@ -18,10 +18,19 @@
 
 $for(author)$
 \address{%
-$author.name$\\
-$author.affiliation$\\
-$for(author.address)$$author.address$$sep$\\ $endfor$\\
+$author.name$
+$for(author.affiliation/pairs)$
```

Comment:
@RLumSK is there something specific to activate with pandoc. This causes an error with me and I don't know why... 🤔 
I get this error 

```
"template" (line 22, column 24):
unexpected "/"
expecting "." or ")$"
```

It happens on my windows 10 computer, but also in Ubuntu 20 (using WSL)

Not sure why... Pandoc Manual explains clearly this but I can't use (https://pandoc.org/MANUAL.html#pipes)

Reply:
I have never used pipes in templates myself.

I checked the release notes of Pandoc, and it seems that the pipes in templates were introduced in v2.8: https://github.com/jgm/pandoc/releases/tag/2.8 (it was originally called "filters" but renamed to "pipes" later: https://github.com/jgm/pandoc/releases/tag/2.9)

Reply:
Oh Ok. That would explain why it passes the CI checks. 
https://github.com/rstudio/rticles/blob/ec5e82b21c89ecb2d62ea935b1a3104fd0e48375/.travis.yml#L7

That means it does not work on RStudio IDE with the shipped version. I think we want it to work.

Reply:
Yep that works with last pandoc version 2.10.1. Thank you! 
Maybe we should make the CI test RStudio version and last version ?

Reply:
I changed the version in 1c5dfcd7a1e2e90b7bec87c9c1b074d340e4f87b because some users reported problems with the `cslreferences` environment, which was created by Pandoc 2.8.

I wonder if there is a way to avoid using pipes here since the current stable release of RStudio ships Pandoc 2.7.x.

Reply:
@yihui @cderv Sorry, I was not aware of the different version shipped with RStudio. However, to address the question of whether this can be done without pipes: Yes, we can modify the template out of R on the fly. This was my first idea,  but it appeared too ugly, so I came up with the pipe solution. However, I was not aware of the RStudio pandoc version issue, but maybe it gets updated also soon in RStudio(?).

Reply:
We need to provide a working template for the current RStudio version.  One solution would be to only use the `for` loop without the pipe, and that would mean affiliation would need to be defined in the Yaml as a list, even for one author.

```yaml
author:
  - name: Author One
    affiliation: 
      - name : Affiliation
      - address:
        - line 1
        - line 2
    url: https://journal.r-project.org
    orcid: 0000-0002-9079-593X
    email:  author1@work
  - name: Author Two
    url: https://journal.r-project.org
    email: author2@work
    orcid: 0000-0002-9079-593X
    affiliation: 
      - name: Affiliation 1
      - address:
        - line 1 affiliation 1
        - line 2 affiliation 1
      - name: Affiliation 2
      - address:
        - line 1 affiliation 2
        - line 2 affiliation 2
```
but this would be a breaking change because the former syntax would not work anymore for only one author. 

We could also add new `affiliation2` and `adress2` fields in the template, so that an author can add another one. This would make it less clean but would make possible the old syntax. And we can change to pandoc pipes for variable when pandocs upgrade in RStudio - in RStudio 1.4, it seems it upgrades to pandoc 2.9.2.1 from the rstudio daily build.

```
$author.affiliation$\\
$for(author.address)$$author.address$$sep$\\ $endfor$\\
}
$author.affiliation2$\\
$for(author.address2)$$author.address2$$sep$\\ $endfor$\\
}
```

How many affiliations one author could be part of for such an article ? Is two enough for most of the case ? Would could make it three without trouble.

Reply:
Maybe we should go for the first list solution. This was also something I had tried, but because it would have broken the old syntax, I did not go for it (again, I was not aware of the RStudio issue). Nevertheless, it is an elegant and still logic way and more important: It probably needs no further update in the near future. If ok for you, I can change the PR accordingly.

Reply:
I would go for the second until a correct pandoc versions is updated in RStudio - I don't think it is a good idea to break this, and that it worth it because I am not sure a lot of authors requires mutliple affiliations as it seems you are the first to ask for it. 
Do you think 2 is not enough for the second solution ?

Reply:
Ok, then we go for solution 2.

Reply:
Done. Please double-check.

Reply:
It seems fine ! I'll thorougly test that and adjust if necessary ! Thank you!